monocle/gitlab/mergerequest.py

```diff
@@ -0,0 +1,545 @@
+            node {
+              __typename
+              ... on ClosedEvent {
+                id
+                createdAt
+                actor {
+                  login
+                }
+              }
+              ... on PullRequestReview {
+                id
+                createdAt
+                state
+                author {
+                  login
+                }
+              }
+              %(force_push_event_item)s
+            }
+          }
+        }
+        author {
+          login
+        }
+        mergedBy {
+          login
+        }
+        repository {
+          owner {
+            login
+          }
+          name
+        }
+      """  # noqa: E501
+        if include_commits and not self.gql.token_getter.can_read_commit():
+            self.log.info(
+                f"Disable commits fetching for group {self.group} due to unsufficent ACLs"
+            )
+            include_commits = False
+        query_params = {
+            "commits": commits if include_commits else "",
+            "force_push_event": force_push_event if include_commits else "",
+            "force_push_event_item": force_push_event_item
+            if include_commits
+            else "",
+        }
+        return pr_query % query_params
+
+    def _getPage(self, kwargs, prs):
+        qdata = """{
+        repository(owner: "%(group)s", name:"%(repository)s") {
+          pullRequests(
+            first: %(size)s
+            %(after)s
+            orderBy: { field: UPDATED_AT, direction: DESC }
+          ) {
+            totalCount
+            pageInfo {
+              hasNextPage endCursor
+            }
+            edges {
+              node {
+                %(pr_query)s
+              }
+            }
+          }
+        }
+      }"""  # noqa: E501
+        data = self.gql.query(qdata % kwargs)
+        if "data" not in data:
+            self.log.error("No data collected: %s" % data)
+            if "message" in data and "wait a few minutes" in data["message"]:
+                self.log.info("sleeping 2 mn")
+                sleep(120)
+            else:
+                self.log.info("sleeping 20 s")
+                sleep(20)
+            return None
+        if not kwargs["total_prs_count"]:
+            kwargs["total_prs_count"] = data["data"]["repository"][
+                "pullRequests"
+            ]["totalCount"]
+            self.log.info(
+                "Total PRs: %s but will fetch until we reached a PR "
+                "updated at date < %s"
+                % (kwargs["total_prs_count"], kwargs["updated_since"])
+            )
+            if kwargs["total_prs_count"] == 0:
+                return False
+        edges = data["data"]["repository"]["pullRequests"]["edges"]
+        for pr in edges:
+            prs.append(pr["node"])
+        oldest_update = sorted(
+            [is8601_to_dt(pr["node"]["updatedAt"]) for pr in edges],
+            reverse=True,
+        )[-1]
```

Comment:
if you are `reverse=True` why take the last one?

you could also negate it and make it clearer maybe?
```suggestion
        oldest_update = sorted(
            [is8601_to_dt(pr["node"]["updatedAt"]) for pr in edges],
        )[0]
```